File: 788. Rotated Digits/_788.py
class Solution:
    def rotatedDigits(self, N: int) -> int:
        count = 0
        for i in range(N+1):
            i = str(i)
            flag = 0
            for j in ["3","4","7"]:
                if j in i:
                    flag = 1
                    break
            if flag == 1:
                continue

            for k in i:
                if k in ["2","5","6","9"]:
                    count+=1
                    break
        return count

    def rotatedDigits1(self, N):
        s1 = set([1, 8, 0])
        s2 = set([1, 8, 0, 6, 9, 2, 5])
        def isGood(x):    
            s = set([int(i) for i in str(x)])
            return s.issubset(s2) and not s.issubset(s1)
        return sum(isGood(i) for i in range(N + 1))

File: 788. Rotated Digits/788.go
package leetcode

import (
	"strconv"
	"strings"
)

func rotatedDigits(N int) int {
	count := 0
	for i := 0; i <= N; i++ {
		is := strconv.Itoa(i)
		if strings.ContainsRune(is, '3') || strings.ContainsRune(is, '4') || strings.ContainsRune(is, '7') {
			continue
		}
		if strings.ContainsRune(is, '2') || strings.ContainsRune(is, '5') || strings.ContainsRune(is, '6') || strings.ContainsRune(is, '9') {
			count++
		}
	}
	return count
}

func rotatedDigits1(N int) int {
	nums := []rune{'0', '1', '5', '-', '-', '2', '9', '-', '8', '6'}
	count := 0
	for i := 1; i <= N; i++ {
		s := strconv.Itoa(i)
		same := 0
		for i, v := range s {
			if nums[v-'0'] == '-' {
				break
			}
			if nums[v-'0'] == v {
				same++
			}
			if same != len(s) && i == len(s)-1 {
				count++
			}
		}
	}
	return count
}
